File: Python/tableinfo.py

class TableInfo(object):
    def __init__(self, conn, name):
        self.name = name
        self.numRows = 0
        self.columns = []
        self.getNumRows(conn)
        self.loadColumns(conn)

    def loadColumns(self, conn):
        cursor = conn.cursor()
        cursor.execute("PRAGMA table_info('" + self.name + "')")
        rows = cursor.fetchall()
        if len(rows) > 0:
            for row in rows:
                self.columns.append((row[0], row[1]))

    def getNumRows(self, conn):
        cursor = conn.cursor()
        cursor.execute("SELECT COUNT(*) FROM '" + self.name + "'")
        row = cursor.fetchone()
        if not row is None:
            self.numRows = row[0]

    def compare(self, table):
        if (self.numRows == table.numRows) and (len(self.columns) == len(table.columns)):
            for index, colInfo in enumerate(self.columns):
                if (colInfo[0] != table.columns[index][0]) or (colInfo[1] != table.columns[index][1]):
                    return False
            return True
        return False

    def toStr(self, includeName):
        if includeName:
            return "{} has {} rows and {} columns".format(self.name, self.numRows, len(self.columns))
        return "has {} rows and {} columns".format(self.numRows, len(self.columns))

File: C#/TableInfo.cs
using System;
using System.Collections.Generic;
using System.Data.Common;
using System.Text;

namespace dbComp
{
    class TableInfo
    {
        String name;
        int numRows = 0;
        List<Tuple<string, string>> columns = new List<Tuple<string, string>>();

        public int NumRows
        {
            get { return numRows; }
        }
        public String Name
        {
            get { return name; }
        }
        public List<Tuple<string, string>> Columns
        {
            get { return columns;  }
        }
        public TableInfo(DbConnection sqlconn, String name)
        {
            this.name = name;
            numRows = getNumRows(sqlconn);
            var command = sqlconn.CreateCommand();
            command.CommandText = "PRAGMA table_info('" + this.name + "')";
            using var reader = command.ExecuteReader();

            while (reader.Read())
            {
                columns.Add(new Tuple<string, string>(reader.GetString(1), reader.GetString(2)));
            }
        }
        int getNumRows(DbConnection sqlconn)
        {
            var command = sqlconn.CreateCommand();
            command.CommandText = "SELECT COUNT(*) FROM '" + this.name + "'";
            using var reader = command.ExecuteReader();
            if (reader.Read())
                return reader.GetInt32(0);
            return 0;
        }
        public bool Compare(TableInfo table)
        {
            if ((numRows == table.numRows) && (columns.Count == table.columns.Count))
            {
                var it = table.columns.GetEnumerator();
                foreach (var colInfo in columns)
                {
                    it.MoveNext();
                    if (!colInfo.Equals(it.Current))
                        return false;
                }
                return true;
            }
            else
                return false;
        }
        public string GetDescription()
        {
            return String.Format("has {0} rows and {1} columns", numRows, columns.Count);
        }
    }
}
